binstar_build_client/worker_commands/su_run.py

```
@@ -0,0 +1,78 @@
+                        absolute_import)
+
+import logging
+import os
+
+from binstar_client.utils import get_binstar
+
+from binstar_build_client import BinstarBuildAPI
+from binstar_client.utils import get_binstar
+
+from binstar_build_client import BinstarBuildAPI
+from binstar_build_client.utils import get_conda_root_prefix
+from binstar_build_client.worker.su_worker import (SuWorker,
+                                                   SU_WORKER_DEFAULT_PATH)
+from binstar_build_client.worker.register import WorkerConfiguration
+from binstar_build_client.worker_commands.run import add_worker_dev_options
+
+log = logging.getLogger('binstar.build')
+
+
+def main(args):
+    worker_config = WorkerConfiguration.load(args.worker_id)
+
+
+    log.info(str(worker_config))
+    worker_home = os.path.expanduser('~{0}'.format(args.build_user))
+    args.conda_build_dir = args.conda_build_dir.format(platform=worker_config.platform,
+                                                       build_user_home=worker_home)
+    log.info("Using conda build directory: {}".format(args.conda_build_dir))
+
+    bs = get_binstar(args, cls=BinstarBuildAPI)
+
+    worker = SuWorker(bs, worker_config, args)
+
+    worker.write_status(True, "Starting")
+
+    try:
+        with worker_config.running():
+            worker.work_forever()
+    finally:
+        worker.write_status(False, "Exited")
+
+
+def add_parser(subparsers, name='su_run',
+               description='Run a build worker to build jobs off of a binstar build queue',
+               epilog=__doc__):
+
+    parser = subparsers.add_parser(name,
+                                   help=description, description=description,
+                                   epilog=epilog
+                                   )
+
+    parser.add_argument('worker_id',
+                        help="worker_id that was given in anaconda build register")
+    parser.add_argument('build_user',
```

@PeterDSteinberg What happens if the user accidentally specifies root? or another user? Is there a differentiator that we can error out in if the user enters an invalid value?

Should we requred the user to **not** exist? Could we run `user-add` to start the script?